src/main/java/org/weasis/dicom/google/api/GoogleAPIClient.java

```diff
@@ -319,40 +319,32 @@ public static String getImageUrl(DicomStore store, String studyId) {
     }
 
     private String formatQuery(StudyQuery query) {
-        String allItems = "?includefield=all";
-        if (query == null) {
-            return allItems;
-        }
-
         List<String> parameters = new ArrayList<>();
-        if (isNotBlank(query.getPatientName())) {
-            parameters.add("PatientName=" + urlEncode(query.getPatientName()));
-        }
-
-        if (isNotBlank(query.getPatientId())) {
-            parameters.add("PatientID=" + urlEncode(query.getPatientId()));
-        }
-
-        if (isNotBlank(query.getAccessionNumber())) {
-            parameters.add("AccessionNumber=" + urlEncode(query.getAccessionNumber()));
-        }
-
-        if (query.getStartDate() != null && query.getEndDate() != null) {
-            parameters.add("StudyDate="
-                    + urlEncode(DATE_FORMAT.format(query.getStartDate()))
-                    + "-" + urlEncode(DATE_FORMAT.format(query.getEndDate()))
-            );
-        }
 
-        if (isNotBlank(query.getPhysicianName())) {
-            parameters.add("ReferringPhysicianName=" + urlEncode(query.getPhysicianName()));
-        }
-
-        if (parameters.isEmpty()) {
-            return allItems;
+        if (Objects.nonNull(query)) {
```

Comment:
I would make the null case the first step to reduce nesting and just return the default if the query was null.

Reply:
are you sure it's good idea to have two return statement in method? it's splits simple logic of forming request string into two

Reply:
Yes but the request string logic is anyway split intp two in the if and else cases...

Reply:
well, this if-else populate list and person who reads may skip this part without losing sense of method
but anyway if you still think two return statement is better i'll revert this changes in one moment

Reply:
Yeah it makes readability much better than the if-else blocks.

Reply:
done

Reply:
Thanks :)